### ISSUE TITLE
Mention specific building type (in building-related quests)

### ISSUE THREAD

**Issue by mnalis** [streetcomplete/StreetComplete#2588]
**Use case**

The _specific building type_ often seems to be wrong (at least here in Croatia). Basically, in vast majority of cases, it is set to `house` on all buildings in residential areas. Quite often, it should be `apartments` instead, and sometimes something else (`commercial` or more specific, or often is abandoned or in ruins in villages and smaller towns). 

In fact, it happens so often, that when SC asks me _"how many levels does this building have"_ I now regularly open a note stating _"apartments, x floors"_ or similar (and then postprocess at home in JOSM) instead of answering the quest, if building is anything but a single family home. In majority of cases, when I get home, I see that it was indeed incorrectly marked `building=house` instead of `building=apartments` or `building=retail` etc.

While the building type is not (IMHO) so important that it would warrant separate error-catching quest (eg. if more than `x` square meters, and more than `y` levels, then it is likely not a single-family `house`, which would like by problematic), it would still be nice to catch those errors.
Also, I do not know how prevalent the problem is in other countries, but judging by how often I see `building`/`building:use` or `detached`/`house` questions, it is likely possible problem exists elsewhere too.

**Proposed Solution**
After `building type` is known, more building related quests pop up, like _"How many levels does this **building** have"_ 

If the `building levels` quests would use **more specific term** instead of (or in addition to) generic **"building"**, then the SC mappers could see if the building type is correctly mapped, and if so, normally answer the question about _building levels_, only leaving notes if they see that the building type is incorrect.

- one option would be to simply extend the question text, eg. _"how many levels does this building (of type **house**) have"_ (where bolded part would be replaced by specific mapped building type, eg: "house", "Semi-detached house", "Shop(s)" etc.)
- or it could be put above the question (like _"house number 42"_ detail does already if housenumber is known)
- alternatively, the icon with building type could be shown - there is code already implemented for that in other places, like:
![sc_building_type](https://user-images.githubusercontent.com/156656/108121382-04f3d180-70a3-11eb-9460-0921f5c6ff0c.jpg)

I'm primarily interested in **building type** info being shown in _"How many levels does this building have"_ quest (as it seems to almost never be mapped in such cases with wrong building type in my experience), but there is no reason why it could not be also added in other building-related quests (like  _"What is the house number of this **building**?"_ etc.) if there is interest for that.

**Comment by westnordost** [comment]
Well, could use the same pattern as for other things...

_"What is the house number of this building? (**Apartment building**)"_ 

...using the id-tagging-presets. Do these presets cover all the buildings that STreetComplete knows and are they translated the same? Would you find this out, @mnalis ?

**Comment by rhhsm** [comment]
mnalis would be helped if there was a quest that would ask for a more detailed building type for buildings that are tagged as house, just like the quest I suggested here #2016

**Comment by peternewman** [comment]
> mnalis would be helped if there was a quest that would ask for a more detailed building type for buildings that are tagged as house, just like the quest I suggested here #2016

Unfortunately house is even more problematic than residential as proposed in #2016 as house is a perfectly acceptable, and indeed the most accurate, answer for some buildings.

I guess there could be an argument for a resurvey of the residential class of properties, and possibly others, every 4 or 8 time periods in case they'd been converted into apartments (common in the UK), or merged back into one big house, but that's not necessarily easy to tell from the outside.

**Comment by rhhsm** [comment]
> 
> 
> > mnalis would be helped if there was a quest that would ask for a more detailed building type for buildings that are tagged as house, just like the quest I suggested here #2016
> 
> Unfortunately house is even more problematic than residential as proposed in #2016 as house is a perfectly acceptable, and indeed the most accurate, answer for some buildings.

It could be implemented by creating a special version of the building type quest that also asks the building type for building=house and building=residential. It should be disabled by default but possible to enable for those who find that in their area, many residential buildings have not been tagged in detail.

> I guess there could be an argument for a resurvey of the residential class of properties, and possibly others, every 4 or 8 time periods in case they'd been converted into apartments (common in the UK), or merged back into one big house, but that's not necessarily easy to tell from the outside.

That should be done by adding a building:use tag.

**Comment by mnalis** [comment]
> Well, could use the same pattern as for other things...
> 
> _"What is the house number of this building? (**Apartment building**)"_
> 
> ...using the id-tagging-presets. Do these presets cover all the buildings that STreetComplete knows and are they translated the same? Would you find this out, @mnalis ?

Yes, I'm willing to put effort if it is within my capabilities (Note that I don't have Java development environment, nor do I speak Java), if someone would point me in right direction.  I can probably use shell or perl for correlation.


But I'm not sure I'm getting this correctly; I though that SC has `quest_buildingType_*` which contained building types and translations, so can't that be used for showing building type like it does in picture at top of this issue (if some rare building type is encountered which is not in SC, we could either print "Other building type" or just show raw value of `building` key). 

It seems to me  however that you imply that SC already uses id-tagging-presets elsewhere ("use the same pattern as for other things...")  (which I assume we talk about [iD](https://wiki.openstreetmap.org/wiki/ID) web-based OSM editor [list of presets](https://github.com/openstreetmap/id-tagging-schema/tree/main/data/presets/building)?) and that you would like to use that instead of StreetComplete own `quest_buildingType_*` ?

If you can confirm that is what you want, then I can try to compare 52 building types that SC seems to use:
```
fgrep '"building' app/src/main/java/de/westnordost/streetcomplete/quests/building_type/BuildingType.kt  | cut -f4 -d\" | sort -u  | wc -l
```

with 55 types iD presets seem to know about:
```
curl -s https://raw.githubusercontent.com/openstreetmap/id-tagging-schema/main/dist/presets.json | grep '"building": "[a-z]' | cut -f4 -d\"| sort -u
``` 

According to that,  building tags which are different are:
```diff -U0 id.tags sc.tags | fgrep -v @
--- id.tags     2021-02-20 19:47:31.722613389 +0100
+++ sc.tags     2021-02-20 19:47:00.514419946 +0100
-barn
-boathouse
-cabin
-cowshed
-entrance
-grandstand
+government
+kiosk
+pagoda
-pavilion
-public
+religious
-ruins
-stable
+shrine
+sports_centre
-sty
+synagogue
+toilets
```

If this is on the right trail of what you want, I assume you also need me to pair that with English string list from SC:
```
egrep -ih 'quest_buildingType_' app/src/main/res/values-en/strings.xml
```

and compare against iD english translations (when I find them), and then show a table with differences? 

Or am I going in the completely wrong direction? :fearful:

**Comment by peternewman** [comment]
> That should be done by adding a building:use tag.

I'm aware that's what the wiki says, although both SC and iD steer you heavily towards usage I'd say "What kind of building is this?". I'd also suggest that apart from for obvious ones like say a church to house, you probably need to be an architectural historian to give the right answer.

> It seems to me however that you imply that SC already uses id-tagging-presets elsewhere ("use the same pattern as for other things...") (which I assume we talk about [iD](https://wiki.openstreetmap.org/wiki/ID) web-based OSM editor [list of presets](https://github.com/openstreetmap/id-tagging-schema/tree/main/data/presets/building)?) and that you would like to use that instead of StreetComplete own `quest_buildingType_*` ?

See https://github.com/westnordost/osmfeatures , specifically https://github.com/westnordost/osmfeatures#get-the-data .

Although it wouldn't have the pretty icons that SC has.

**Comment by mnalis** [comment]
Hi @westnordost, I've created this [gist](https://gist.github.com/mnalis/cecbd88d41be83f12801a34cdec734fb) and it says those are differences between StreetComplete building type English translations, and iD presets English translations (for `building` tags whose description differ in more than just in lower/uppercase):
 
Tag | SC | iD
--- | --- | ---
barn||Barn
boathouse||Boathouse
cabin||Cabin
cathedral|Cathedral|Cathedral Building
chapel|Chapel|Chapel Building
church|Church|Church Building
construction||Building Under Construction
cowshed||Cowshed
entrance||Entrance/Exit
farm|On a farm|Farm House
garage|Single garage|Garage
garages|Multiple garages|Garages
government|Government building|
grandstand||Grandstand
hangar|Hangar|Hangar Building
kindergarten|Kindergarten building|Preschool / Kindergarten Building
kiosk|Kiosk|
mosque|Mosque|Mosque Building
pagoda|Pagoda|
parking|Parking building|
pavilion||Pavilion Building
public||Public Building
religious|Religious building|
residential|Residential|Residential Building
retail|Shop(s)|Retail Building
ruins|Ruins|Building Ruins
semidetached_house||Semi-Detached House
shrine|Shrine|
sports_centre|Sports centre|
stable||Stable
stadium|Stadium|Stadium Building
static_caravan|Mobile home|Static Mobile Home
sty||Pigsty
synagogue|Synagogue|
terrace||Row Houses
toilets|Toilets|
train_station|Train station|Train Station Building
transportation|Public transportation building|Transportation Building

But as I said before:

> But I'm not sure I'm getting this correctly; I though that SC has quest_buildingType_* which contained building types and translations, so can't that be used for showing building type like it does in picture at top of this issue? (if some rare building type is encountered which is not in SC, we could either print "Other building type" or just show raw value of building key)

**Comment by westnordost** [comment]
Hmm, thanks, nevermind then. The app should not show the iD names when they are so different. So this requires to implement it a little different.

**Comment by westnordost** [comment]
Hm, I started implementing this, but noticed that this cannot be done until the view-part of quests is separated from the rest

#2057

**Comment by peternewman** [comment]
@westnordost I notice you've closed the linked issue #2057 . Presumably that should make this either unblocked or closed depending on whether #2057 was closed as not possible to do, or just during some clean-up (I don't believe it's been done).
